app/controllers/docker_manager/admin_controller.rb

```diff
@@ -174,5 +180,14 @@ def repo_version(repo)
     def concat_repos_versions(repos)
       repos.map(&:latest_local_commit).join(", ")
     end
+
+    def discourse_upgrade_required?(min_stable_version, min_beta_version)
+      tracking_stable = Discourse::VERSION::PRE.nil?
+      discourse_version = Gem::Version.new(Discourse.full_version.delete_prefix('v'))
```

Comment:
```suggestion
      discourse_version = Gem::Version.new(Discourse::VERSION::STRING)
```

Maybe relying on that constant will be a bit more reliable? `Discourse.full_version` does a `git describe` command.